### PR TITLE
Key-pair auth for Cortex Provider

### PR DESCRIPTION
**Pull request by sfc-gh-jreini**
# Description

Key-pair auth for Cortex Provider. Enabled from env variables (same as user/password)

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/benchmark/trulens/benchmark/benchmark_frameworks/tru_benchmark_experiment.py [hunk 1]
```diff
@@ -173,4 +172,3 @@ def create_benchmark_experiment_app(
         app_id=app_id,
         feedbacks=benchmark_experiment.f_benchmark_metrics,
         **kwargs,
```
Comment (sfc-gh-pdharmana) [inline src/benchmark/trulens/benchmark/benchmark_frameworks/tru_benchmark_experiment.py:174]: Unintended change?

File: src/providers/cortex/trulens/providers/cortex/provider.py [hunk 2]
```diff
@@ -32,15 +34,47 @@ def __init__(
...
+            "schema": os.environ["SNOWFLAKE_SCHEMA"],
+            "warehouse": os.environ["SNOWFLAKE_WAREHOUSE"],
+        }
+
+        if "SNOWFLAKE_USER_PASSWORD" in os.environ:
+            connection_parameters["password"] = os.environ[
+                "SNOWFLAKE_USER_PASSWORD"
+            ]
+
+        if "SNOWFLAKE_PRIVATE_KEY" in os.environ:
```
Comment (sfc-gh-pdharmana) [inline src/providers/cortex/trulens/providers/cortex/provider.py:52]: `elif`? Also should this order be reversed? First give priority to key based auth?

File: src/providers/cortex/trulens/providers/cortex/provider.py [hunk 3]
```diff
@@ -32,15 +34,47 @@ def __init__(
...
+            # Retrieve the private key as a string from the environment
```
Comment (sfc-gh-pdharmana) [inline src/providers/cortex/trulens/providers/cortex/provider.py:53]: Add a log line to show we're going with key based auth.

**Comment by sfc-gh-pdharmana** [comment]
Too many unintended changes?

**Comment by sfc-gh-pdharmana** [comment]
@sfc-gh-jreini also, I thought we cannot read the pem file in streamlit and we would directly read from env file. Would we support both cases, if yes should that be part of the same PR?

File: src/dashboard/setup.py [hunk 4]
```diff
@@ -13,7 +13,6 @@
 
 from setuptools import setup
```
Comment (sfc-gh-chu) [inline src/dashboard/setup.py:14]: I think we need to revert the changes to `src/dashboard`
